### PR TITLE
Slac master

### PR DESCRIPTION
**Pull request by bhill-slac**
First round of fixes for adl2edl.
Provides more reasonable defaults and fixes bugs in embedded windows, related displays and rgb mode.

### CODE REVIEW

File: control.cc [hunk 1]
```diff
@@ -661,7 +661,11 @@ int relatedclass::parse(ifstream &inf, ostream &outf, ostream &outd)
 				outd << "START " << tname << endl;
 				saved_name = tname;
 
-				if(tname.length() == 0) { // medm ignores blank names. edm does not.
+				if(tname.length() != 0) { // medm ignores blank names. edm does not.
+					adl_pos = tname.find(adl,0);
+					if(adl_pos != -1){
+						tname.replace(adl_pos,4,echar);
+					}
```
Comment (mdavidsaver) [inline control.cc]: I won't pretend to understand what this code does (or did).  Is this just stripping any '.adl' suffix?  All the other changes look straightforward.

**Comment by mdavidsaver** [comment]
@ralphlange Will you have time to review/merge this?

**Comment by ralphlange** [comment]
As I haven't been using neither medm nor edm for four years, I don't really feel appropriate, I have little justification to spend work hours on it, and I have no way to test.

So - I can look at the code, but that's about all.

File: control.cc (hunk 1 above)
Comment (ralphlange) [reply control.cc]: It is replacing an ".adl" suffix with ".edl" - which sounds very reasonable for adl2edl.

**Comment by ralphlange** [comment]
Changes look good to me.
(Disclaimer: reviewed by solely looking at the changes, no compilation, no tests.)

**Comment by ralphlange** [comment]
Merge now and do the Debian related updating later? Who is going to prepare a new Debian version?
Martin? Joachim?         (@mark0n @jrcoding)

**Comment by mark0n** [comment]
At FRIB, we are not using this package anymore. Maybe someone else wants to prepare a new Debian version? I would be happy to help (if help is needed).

**Comment by mdavidsaver** [comment]
adl2edl is currently used during the build phase for the s7plc and areadetector packaging.  Also the now effectively deprecated synapps package.  The AD package in particular will benefit as these are among the screens being fixed.

That said, I don't really want to deal with adl2edl anymore, and it seems like no one else does either.  So I'm going to grant @bhill-slac commit privileges to this repo, thus bestowing on him the much coveted title of Maintainer of adl2edl (my condolences Bruce ;) ).

**Comment by ralphlange** [comment]
Is this repo both packaging and upstream?

**Comment by mdavidsaver** [comment]
As far as I'm concerned, yes by default (is there another repo/maintainer?).

Also, I don't have any expectation that Bruce will update the packaging files under 'debian/'.  I'm happy that he's bothering to fix this long standing issue.

**Comment by bhill-slac** [comment]
Thanks for the push access!  A privilege and a burden!
Re debian packaging support, I'm not familiar w/ debian and
don't have access to a test machine, but it sounds like help is
available if needed.

I'll push these changes as they've survived review and my testing.
I have some more WIP so will hold off on a new tag till I get that pushed.
Cheers, - Bruce

**Comment by ralphlange** [comment]
No worries.
Tell me when you think the stuff is stable and tested, and I'll use what's left of my Debian-fu to update the packaging.

**Pull request merged**